src/storage/src/hummock/compactor/mod.rs

```diff
@@ -223,11 +225,38 @@ impl Compactor {
         let multi_filter_key_extractor = Arc::new(multi_filter_key_extractor);
 
         let mut task_status = TaskStatus::Success;
-        if let Err(e) = generate_splits(&mut compact_task, context.clone()).await {
-            tracing::warn!("Failed to generate_splits {:#?}", e);
-            task_status = TaskStatus::ExecuteFailed;
-            Self::compact_done(&mut compact_task, context.clone(), vec![], task_status).await;
-            return task_status;
+        // skip sst related to non-existent able_id to reduce io
+        let sstable_infos = compact_task
+            .input_ssts
+            .iter()
+            .flat_map(|level| level.table_infos.iter())
+            .filter(|table_info| {
+                let table_ids = &table_info.table_ids;
+                table_ids
+                    .iter()
+                    .any(|table_id| existing_table_ids.contains(table_id))
+            })
+            .cloned()
+            .collect_vec();
+        let compaction_size = compact_task
```

Comment:
we only need to calculate the size of sstables which not be filtered

Reply:
fix it